airflow_dbt_python/hooks/remote/gcs.py

```diff
@@ -0,0 +1,234 @@
+            )
+        except ValueError:
+            success = False
+            self.log.warning("Failed to load %s: key already exists in GCS.", key)
+
+        return success
+
+    def _download(
+        self,
+        source: URL,
+        destination: URL,
+        replace: bool = False,
+        delete_before: bool = False,
+    ):
+        """Download one or more files from a destination URL in GCS.
+
+        Lists all GCS keys that have source as a prefix to find what to download.
+
+        Args:
+            source: An GCS URL to a key prefix containing objects to download.
+            destination: A destination URL where to download the objects to. The
+                existing sub-directory hierarchy in GCS will be preserved.
+            replace: Indicates whether to replace existing files when downloading.
+                This flag is kept here to comply with the DbtRemote interface but its
+                ignored as files downloaded from GCS always overwrite local files.
+            delete_before: Delete destination directory before download.
+        """
+        gcs_object_keys = self.iter_url(source)
+
+        if destination.exists() and delete_before is True:
+            for _file in destination:
+                _file.unlink()
+
+            if destination.is_dir():
+                destination.rmdir()
+
+        for gcs_object_key in gcs_object_keys:
+            self.log.info("GCSObjectKey: %s", gcs_object_key)
+            self.log.info("Source: %s", source)
+
+            bucket_name, object_name = _parse_gcs_url(str(gcs_object_key))
+            gcs_object = self.get_key(object_name, bucket_name)
+            gcs_object_url = URL(gcs_object_key)
+
+            if source != gcs_object_url and gcs_object_url.is_relative_to(source):
+                gcs_object_url = gcs_object_url.relative_to(source)
+
+            if gcs_object_url.suffix == "" and str(gcs_object_url).endswith("/"):
+                # Empty GCS files may also be confused with unwanted directories.
+                self.log.warning(
+                    "A file with no name was found in GCS at %s", gcs_object
+                )
+                continue
+
+            if destination.is_dir():
+                destination_url = destination / gcs_object_url.path
+            else:
+                destination_url = destination
+
+            destination_url.parent.mkdir(parents=True, exist_ok=True)
+
+            gcs_object.download_to_filename(str(destination_url))
+
+    def iter_url(self, source: URL) -> Iterable[URL]:
+        """Iterate over an GCS key given by a URL."""
+        bucket_name, key_prefix = _parse_gcs_url(str(source))
+
+        for key in self.list(bucket_name=bucket_name, prefix=key_prefix):
+            if key.endswith("//"):
+                # Sometimes, GCS files with empty names can appear, usually when using
+                # the UI. These empty GCS files may also be confused with directories.
+                continue
```

Comment:
This same comment is found in the S3 hook. Does GCS also has the same behavior or was it just a copy pasted here?

Reply:
It's copy-paste, I'll clean it up